### PR TITLE
Bundling chkware as zipapp documentation

### LINKED ISSUE #132: build `zipapp`
**What**

build a `zipapp` package that 
- requires no extra downloading on dependent packages
- user just download the zipapp and use as `python3 chk.pyz` (for 1st version of this zip app build)

**Expected outcome**
- documentation of how to create one containing list of steps

### PR DESCRIPTION
**Pull request by sudiptob2**
closes #132

### CODE REVIEW

File: docs/ZIPAPP_BUILD_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+## Creating Zipapp using [shiv](https://shiv.readthedocs.io/en/latest/)
+Shiv is a command line utility for building fully self-contained Python zipapps as outlined in PEP 441, but with all their dependencies included.
+It helps us to handle all those corner cases that might cause trouble while creating zipapp.
+
+### Installation
+
+```bash
+pip install shiv
```
Comment (0hsn) [inline docs/ZIPAPP_BUILD_GUIDE.md:25]: why `pip` and not `pipenv`?

File: docs/ZIPAPP_BUILD_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+`python3` is the python interpreter that will be used to bundle `chkware` as zipapp.
+
+`shiv` is the command itself.
+
+`-c chk` specifies the `console_script` for chkware ([defined here](https://github.com/chkware/cli/blob/f21698b1242e20b6bafbb11c597b455088f94de3/setup.cfg#L73))
+
+`-o chk.pyz` specifies the `outfile`
+
+`./cli` specifies the folder where the project resources are stored.
```
Comment (0hsn) [inline docs/ZIPAPP_BUILD_GUIDE.md:44]: you can just put a link to the reference doc.

File: docs/ZIPAPP_BUILD_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+### Installation
+
+```bash
+pip install shiv
+```
+### Bundling zipapp
+From the terminal run following command to bundle `chkware` as zipapp.
+
+```bash
+python3 -m shiv -c chk -o chk.pyz ./cli
```
Comment (0hsn) [inline docs/ZIPAPP_BUILD_GUIDE.md:31]: - use with pipenv 
- write full create process, from including `git clone` to build

File: docs/ZIPAPP_BUILD_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+`shiv` is the command itself.
+
+`-c chk` specifies the `console_script` for chkware ([defined here](https://github.com/chkware/cli/blob/f21698b1242e20b6bafbb11c597b455088f94de3/setup.cfg#L73))
+
+`-o chk.pyz` specifies the `outfile`
+
+`./cli` specifies the folder where the project resources are stored.
+
+## Conclusion
```
Comment (0hsn) [inline docs/ZIPAPP_BUILD_GUIDE.md:46]:
```suggestion
## Caveats
```

File: docs/ZIPAPP_BUILD_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Zipapps are not guaranteed to be cross-compatible with different **OS** architectures. 
+For example, a pyz file built on a Mac may only work on other Macs, likewise for RHEL, etc. 
+This usually only applies to zipapps that have C extensions in their dependencies. 
+If all your dependencies are pure python, then chances are the pyz will work on other platforms.
```
Comment (0hsn) [inline docs/ZIPAPP_BUILD_GUIDE.md:51]: these are generic texts, write something that actually users need to do perform specific (if they have to) build for themself

File: docs/ZIPAPP_BUILD_GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+Zipapps created with shiv will extract themselves into `~/.shiv`, unless overridden via `SHIV_ROOT`. 
+If you create many utilities with shiv, you may want to occasionally clean this directory.
```
Comment (0hsn) [inline docs/ZIPAPP_BUILD_GUIDE.md:54]: these are generic texts, write something that actually users need to do perform specific (if they have to) build for themself

**Review by 0hsn — CHANGES_REQUESTED** [review]
- Update file name to `docs/ziapp-build-guide.md`
- Add a `docs/ziapp-user-guide.md` on how to download, and use the zipapp [will delete this, and add the doc to website]
- Think about all the, text you wrote, those are true for all zipapp, but you need to write based on `chkware` only.

File: docs/ZIPAPP_BUILD_GUIDE.md (hunk 1 above)
Comment (sudiptob2) [reply docs/ZIPAPP_BUILD_GUIDE.md:25]: If anyone wants to install `shiv` inside the projects `virtualenv` then he can use `pipenv`. However, if you just want to make a zipapp of `chkware` and do not want to set up python virtualenv for development then I guess just installing `shiv` using pip is OK.

Comment (sudiptob2) [reply docs/ZIPAPP_BUILD_GUIDE.md:25]: Understood

**Comment by 0hsn** [comment]
@sudiptob2 let's follow this format https://github.com/chkware/cli/pull/139 🙏

**Review by 0hsn — APPROVED** [review]
I didn't tried to create a zipapp by using this doc. Hope you did. 🤞

**Comment by sudiptob2** [comment]
> I didn't tried to create a zipapp by using this doc. Hope you did. crossed_fingers

Yes, I have tested the steps and am able to create a zipapp following the steps.

**Pull request merged**